### PR TITLE
OpenJDK support, error handling, no Docker, sources vs jars

### PR DESCRIPTION
**Pull request by alexsaveliev**
- Dealing with OpenJDK projects
    Introduced Sourcefile generator for OpenJDK projects. Expected scenario
     - clone OpenJDK
     - build OpenJDK
     - generate Sourcefile for jdk, langtools, and nashorn
     - run `srclib config && srclib make`

  Documented steps required to index OpenJDK
- Made OpenJDK source unit names not-empty
- When reporting compilation errors, including current unit information
- Improved error handling when extracting metadata
    - improved error handling when extracting metadata - indicating place where error occurred in the log
    - silently handling exceptions thrown by .getResoledArtifacts() and derived .asPath()
    - formatted code
- Removed Docker remnants
- Updated java-gradle-sample and java-android-gradle-sample test scenario to produce stable output disregarding Gradle version used by adding Gradle wrapper
- When there is external dependency for which there is local source unit, adding unit's source directories to compiler arguments (picking sources rather than library)

### CODE REVIEW

**Comment by beyang** [comment]
Should the `Srcfile` generation work on the test case `github.com/sgtest/java-jdk-sample`? If so, can we get rid of the `Srcfile` in that repository?

File: src/main/java/com/sourcegraph/javagraph/JDKSrcFileGenerator.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * - there is no need to run scanners, everything is included into Srcfile's units
+ * - source units should contain special markers indicating that we are working with OpenJDK project
+ * Usage: JDKSrcFileGenerator PROJECT [DIRECTORY]
+ * Expected scenario
+ * - clone OpenJDK
+ * - build OpenJDK
+ * - generate Sourcefile for jdk, langtools, and nashorn
+ * - run `srclib make`
+ */
+public class JDKSrcFileGenerator {
```
Comment (beyang) [inline src/main/java/com/sourcegraph/javagraph/JDKSrcFileGenerator.java]: Add a unit test for this?

File: src/main/resources/metainfo.gradle [hunk 2]
```diff
@@ -193,8 +199,8 @@ allprojects {
                             if (isConfigSupported(conf)) {
                                 println "SRCLIB-DEPENDENCY $conf.name:$group:$name:$version:$file"
                             }
-                        } catch (Exception e) {
-                            println "SRCLIB-WARNING $e"
+                        } catch (e) {
+                            println "SRCLIB-WARNING 1 $e"
```
Comment (beyang) [inline src/main/resources/metainfo.gradle]: Putting the numbers after SRCLIB-WARNING seems a little hacky. Also, maybe "Warning: $e" instead?

**Comment by beyang** [comment]
minor comments, but LGTM otherwise

File: src/main/java/com/sourcegraph/javagraph/JDKSrcFileGenerator.java (hunk 1 above)
Comment (alexsaveliev) [inline src/main/java/com/sourcegraph/javagraph/JDKSrcFileGenerator.java]: Added unit test

File: src/main/resources/metainfo.gradle (hunk 2 above)
Comment (alexsaveliev) [inline src/main/resources/metainfo.gradle]: Replaced with exception stack traces (I'd like to be able to understand the place where exception occurred)

**Comment by alexsaveliev** [comment]
> Should the Srcfile generation work on the test case github.com/sgtest/java-jdk-sample? 

Yes, it may work there

> If so, can we get rid of the Srcfile in that repository?

I think we can't - in order to scan repository for source units we should give a hint that we are scanning OpenJDK-alike repository, thus we should already have `Srcfile`

**Pull request merged**